doc/cache_layout.md

```diff
@@ -0,0 +1,243 @@
+# The Pub cache
+
+The Pub cache is where pub stores downloaded packages and globally activated
+packages.
+
+The information in this document is informational, and can be used for
+understanding the cache, but we strongly encourage all manipulation of the cache
+happens though the `dart pub`/`flutter pub` commands to avoid relying on
+accidental properties of the cache that might be broken in the future.
+
+See [system_cache](../lib/src/system_cache.dart) for implementation of top-level
+cache conventions.
+
+## Location
+
+For the Dart SDK the default pub cache location is `$HOME/.pub_cache` on Linux
+and Mac OS, and `%LOCALAPPDATA%/Pub/Cache` on Windows.
+
+For the Flutter SDK after 3.8 uses the same folder. Before it would use the same
+folder if it was found, and otherwise  use `$FLUTTER_ROOT/.pub_cache`.
+
+The environment variable `PUB_CACHE` can be used to override the location of the
+pub cache.
```

Comment:
The default location of the pub-cache can be overridden using the environment variable `PUB_CACHE`.